src/microbe_stage/MicrobeStage.cs

```diff
@@ -712,31 +690,21 @@ protected override void UpdatePatchSettings(bool promptPatchNameChange = true)
         UpdatePatchLightLevelSettings();
     }
 
-    protected override void Dispose(bool disposing)
+    protected override void OnLightLevelUpdate()
     {
-        if (disposing)
-        {
-            GuidanceLinePath?.Dispose();
-        }
-
-        base.Dispose(disposing);
-    }
+        if (GameWorld.Map.CurrentPatch == null)
+            return;
 
-    private void UpdateBackground()
-    {
-        Camera.SetBackground(SimulationParameters.Instance.GetBackground(
-            GameWorld.Map.CurrentPatch!.BiomeTemplate.Background));
-    }
+        patchManager.UpdatePatchBiome(GameWorld.Map.CurrentPatch);
+        patchManager.UpdateAllPatchLightLevels();
```

Comment:
Doesn't have to happen now, but could you add a TODO here that it would make more sense for the GameWorld to update its patch map data based on the light cycle in it. That'd be much clearer OOP design where not everything is messing with the internals of that object.

Reply:
Added.